cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -475,6 +479,40 @@ public static KafkaCluster fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup vers
             if (listeners.getTls() != null && listeners.getTls().getAuth() != null && listeners.getTls().getAuth() instanceof KafkaListenerAuthenticationOAuth) {
                 validateOauth((KafkaListenerAuthenticationOAuth) listeners.getTls().getAuth(), "TLS listener");
             }
+
+            if (listeners.getExternal() != null) {
+                if (result.isExposedWithIngress()) {
+                    if (((KafkaListenerExternalIngress) listeners.getExternal()).getConfiguration() != null &&
+                            ((KafkaListenerExternalIngress) listeners.getExternal()).getConfiguration().getServerKey() != null) {
+                        result.setSecretSource(((KafkaListenerExternalIngress) listeners.getExternal()).getConfiguration().getServerKey());
+                    }
+                }
+
+                if (result.isExposedWithNodePort()) {
+                    if (((KafkaListenerExternalNodePort) listeners.getExternal()).getConfiguration() != null &&
+                            ((KafkaListenerExternalNodePort) listeners.getExternal()).getConfiguration().getServerKey() != null) {
+                        result.setSecretSource(((KafkaListenerExternalNodePort) listeners.getExternal()).getConfiguration().getServerKey());
+                    }
+                }
+
+                if (result.isExposedWithLoadBalancer()) {
+                    if (((KafkaListenerExternalLoadBalancer) listeners.getExternal()).getConfiguration() != null &&
+                            ((KafkaListenerExternalLoadBalancer) listeners.getExternal()).getConfiguration().getServerKey() != null) {
+                        result.setSecretSource(((KafkaListenerExternalLoadBalancer) listeners.getExternal()).getConfiguration().getServerKey());
+                    }
+                }
+
+                if (result.isExposedWithRoute()) {
+                    if (((KafkaListenerExternalRoute) listeners.getExternal()).getConfiguration() != null &&
+                            ((KafkaListenerExternalRoute) listeners.getExternal()).getConfiguration().getServerKey() != null) {
+                        result.setSecretSource(((KafkaListenerExternalRoute) listeners.getExternal()).getConfiguration().getServerKey());
+                    }
+                }
+            }
+
+            if (listeners.getTls() != null && listeners.getTls().getConfiguration() != null) {
+                result.setSecretSource(listeners.getTls().getConfiguration().getServerKey());
```

Comment:
You have separate configuration of the key for `external` and `tls`. But you use the same `secretSource` field here, so they will overwrite each other.